controllers/content_items_controller.go

```diff
@@ -30,9 +30,19 @@ func (c *ContentItemsController) PutDraftContentItem(w http.ResponseWriter, r *h
 
 func (c *ContentItemsController) PutLiveContentItem(w http.ResponseWriter, r *http.Request) {
 	registerWithURLArbiterAndForward(c.arbiter, w, r, func(basePath string, requestBody []byte) {
-		// TODO: PUT to both content stores concurrently
-		doContentStoreRequest(c.liveContentStore, "PUT", basePath, requestBody, w)
-		// for now, we ignore the response from draft content store for storing live content, hence `w` is nil
-		doContentStoreRequest(c.draftContentStore, "PUT", basePath, requestBody, nil)
+		done := make(chan bool)
+		go func() {
+			doContentStoreRequest(c.liveContentStore, "PUT", basePath, requestBody, w)
+			done <- true
+		}()
+		go func() {
+			// for now, we ignore the response from draft content store for storing live content, hence `w` is nil
+			doContentStoreRequest(c.draftContentStore, "PUT", basePath, requestBody, nil)
+			done <- true
+		}()
+		// wait till both requests are forwarded
+		// https://gobyexample.com/channel-synchronization
+		<-done
+		<-done
```

Comment:
I think a [`sync.WaitGroup`](http://golang.org/pkg/sync/#WaitGroup) would be a better primative to use here...

You could also then also `defer` the call to `wg.Done()` at the top of the goroutines to prevent this deadlocking if something panics.